nomos-sdp/src/lib.rs

```diff
@@ -0,0 +1,234 @@
+pub mod state;
+
+use std::{
+    collections::{HashMap, HashSet},
+    hash::Hash,
+};
+
+use blake2::{Blake2b, Digest};
+use multiaddr::Multiaddr;
+
+pub type StakeThreshold = u64;
+pub type BlockNumber = u64;
+
+pub struct MinStake {
+    pub threshold: StakeThreshold,
+    pub timestamp: BlockNumber,
+}
+
+#[derive(Clone, Debug)]
+pub struct ServiceParameters<ContractAddress> {
+    pub lock_period: u64,
+    pub inactivity_period: u64,
+    pub retention_period: u64,
+    pub reward_contract: ContractAddress,
+    pub timestamp: BlockNumber,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+pub struct Locator {
+    addr: Multiaddr,
+}
```

Comment:
Should this be a wrapper type?
In fact, does this multiaddress matches the restrictions from the spec (in terms of size)?
> The length of a regular libp2p multiaddr is 83 characters in the longest case. However, if we allow to Dynamic DNS for retrieving dynamically set global IP addresses for some nodes, then the longest theoretical DNS name is 253 characters. Now assuming that this lengthy DNS must be part of the multiaddr, then we get 329 characters in the longest case.